File: 1071-greatest-common-divisor-of-strings/1071-greatest-common-divisor-of-strings.java
class Solution {
       public String gcdOfStrings(String str1, String str2) {
        if (!(str1 + str2).equals(str2 + str1)) return "";
        int gcd = gcd(str1.length(), str2.length());
        return str2.substring(0, gcd);
    }

    int gcd(int s1, int s2) {
        return s2 == 0 ? s1 : gcd(s2, s1 % s2);
    }
}

File: 1071-greatest-common-divisor-of-strings/1071-greatest-common-divisor-of-strings.kt
class Solution {
    fun gcdOfStrings(str1: String, str2: String): String {
        // Check if gcd exists
        if (str1 + str2 != str2 + str1) {
            return ""
        }
        
        // Count gcd of string lengths
        var l1 = str1.length
        var l2 = str2.length
        while (l1 != l2) {
            if (l1 > l2) l1 -= l2
            else l2 -= l1
        }

        return str1.substring(0, l1)
    }
}